queue/Queue.java

```diff
@@ -0,0 +1,115 @@
+import java.util.NoSuchElementException;
```

Comment:
Either import `java.util.Queue` or don't use `implements Queue<T>`. 

*Note*: If you decide to import `java.util.Queue`, then override `clear` method also.

Reply:
Agree - this was a leftover from work i was doing to implement queue in different ways. Removed